tests/contrib/pytest/test_pytest.py

```diff
@@ -394,3 +410,32 @@ def __repr__(self):
     decoded = json.loads(encoded)
     assert decoded["b"] == repr(dict_side_effect)
     assert decoded["c"] == repr(repr_side_effect)
+
+
+@pytest.mark.parametrize(
+    "repository_url,repository_name",
+    [
+        ("https://github.com/DataDog/dd-trace-py.git", "dd-trace-py"),
+        ("https://github.com/DataDog/dd-trace-py", "dd-trace-py"),
+        ("git@github.com:DataDog/dd-trace-py.git", "dd-trace-py"),
+        ("git@github.com:DataDog/dd-trace-py", "dd-trace-py"),
+        ("dd-trace-py", "dd-trace-py"),
+        ("git@hostname.com:org/repo-name.git", "repo-name"),
+        ("git@hostname.com:org/repo-name", "repo-name"),
+        ("ssh://git@hostname.com:org/repo-name", "repo-name"),
+        ("git+git://github.com/org/repo-name.git", "repo-name"),
+        ("git+ssh://github.com/org/repo-name.git", "repo-name"),
+        ("git+https://github.com/org/repo-name.git", "repo-name"),
```

Comment:
Can we have some invalid string test cases as well to ensure potential errors are handled? Might also be worth asserting the log message is generated too to aid in debugging

Reply:
Do you mean like in `test_repository_name_not_extracted_warning()`?

Reply:
Oh, missed that haha! Yes, like that 😄. Maybe a few more cases but I'm good with it as-is. :+1: